orchagent/isolationgrouporch.cpp

```diff
@@ -0,0 +1,754 @@
+            status = sai_port_api->set_port_attribute((port.m_type == Port::PHY ? port.m_port_id : port.m_lag_id), &attr);
+            if (SAI_STATUS_SUCCESS != status)
+            {
+                SWSS_LOG_ERROR("Unable to set attribute %d value  0x%" PRIx64 "to %s",
+                               attr.id,
+                               attr.value.oid,
+                               port.m_alias.c_str());
+            }
+            else
+            {
+                m_bind_ports.push_back(port.m_alias);
+            }
+        }
+        else
+        {
+            m_pending_bind_ports.push_back(port.m_alias);
+            SWSS_LOG_NOTICE("Port %s saved in pending bind ports for isolation group %s of type %d",
+                            port.m_alias.c_str(),
+                            m_name.c_str(),
+                            m_type);
+        }
+    }
+    else
+    {
+        return ISO_GRP_STATUS_INVALID_PARAM;
+    }
+
+    return ISO_GRP_STATUS_SUCCESS;
+}
+
+isolation_group_status_t
+IsolationGroup::unbind(Port &port, bool do_fwd_ref)
+{
+    SWSS_LOG_ENTER();
+    sai_attribute_t attr;
+    sai_status_t status = SAI_STATUS_SUCCESS;
+
+    if (find(m_bind_ports.begin(), m_bind_ports.end(), port.m_alias) == m_bind_ports.end())
+    {
+        auto node = find(m_pending_bind_ports.begin(), m_pending_bind_ports.end(), port.m_alias);
+        if (node != m_pending_bind_ports.end())
+        {
+            m_pending_bind_ports.erase(node);
+        }
+
+        return ISO_GRP_STATUS_SUCCESS;
+    }
+
+    attr.value.oid = SAI_NULL_OBJECT_ID;
+    if (m_type == ISOLATION_GROUP_TYPE_BRIDGE_PORT)
+    {
+        attr.id = SAI_BRIDGE_PORT_ATTR_ISOLATION_GROUP;
+        status = sai_bridge_api->set_bridge_port_attribute(port.m_bridge_port_id, &attr);
+    }
+    else if (m_type == ISOLATION_GROUP_TYPE_PORT)
+    {
+        attr.id = SAI_PORT_ATTR_ISOLATION_GROUP;
+        status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
+    }
+    else
+    {
+        return ISO_GRP_STATUS_INVALID_PARAM;
+    }
+
+    if (SAI_STATUS_SUCCESS != status)
+    {
+        SWSS_LOG_ERROR("Unable to set attribute %d value 0x%" PRIx64 "to %s", attr.id, attr.value.oid, port.m_alias.c_str());
+    }
+    else
+    {
+        m_bind_ports.erase(find(m_bind_ports.begin(), m_bind_ports.end(), port.m_alias));
+    }
+
+    if (do_fwd_ref)
+    {
+        m_pending_bind_ports.push_back(port.m_alias);
+    }
+
+    return ISO_GRP_STATUS_SUCCESS;
+}
+
+isolation_group_status_t
+IsolationGroup::setBindPorts(string ports)
+{
+    SWSS_LOG_ENTER();
+    vector<string> old_bindports = m_pending_bind_ports;
+    auto port_list = tokenize(ports, ',');
+    set<string> portList(port_list.begin(), port_list.end());
+
+    old_bindports.insert(old_bindports.end(), m_bind_ports.begin(), m_bind_ports.end());
+    for (auto alias : portList)
+    {
+        if ((0 == alias.find("Ethernet")) || (0 == alias.find("PortChannel")))
```

Comment:
Does it mean we can try to bind to LAG? From what I know, there is no `SAI_LAG_ATTR_ISOLATION_GROUP` attribute. So, to bind on LAG we should iterate over LAG members. But `::bind()` does not have this logic. Could you please elaborate more on this?

Reply:
@akokhan  Thanks for the comment. The LAG support is added considerint future support. We are removing the LAG specific code for now as the support is not there.